### PR TITLE
feat: combine infrastructure

### PR DESCRIPTION
**Pull request by bruuuuuuuce**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -14,10 +22,19 @@ precommit: format lint
 run-chatbot:
 	@poetry run streamlit run robo_mo/chatbot.py
 
+python-install:
+	poetry install
```
Comment (bruuuuuuuce) [inline Makefile:26]:
```suggestion
	pip3 install poetry && poetry install
```

File: Makefile [hunk 2]
```diff
@@ -14,10 +22,19 @@ precommit: format lint
 run-chatbot:
 	@poetry run streamlit run robo_mo/chatbot.py
 
+python-install:
+	pip3 install poetry && poetry install
```
Comment (bruuuuuuuce) [inline Makefile:26]:
```suggestion
	pip install poetry && poetry install
```

File: Makefile [hunk 3]
```diff
@@ -14,10 +22,19 @@ precommit: format lint
 run-chatbot:
 	@poetry run streamlit run robo_mo/chatbot.py
 
+python-install:
+	pip install poetry && poetry install
```
Comment (bruuuuuuuce) [inline Makefile:26]:
```suggestion
	pipx install poetry && poetry install
```

File: Makefile [hunk 4]
```diff
@@ -16,8 +24,14 @@ run-chatbot:
...
+	docker build -t robo-mo -f Dockerfile .
 
 .PHONY: docker-run
 docker-run:
 	@docker run -p 8501:8501 -e MOMENTO_API_KEY=${MOMENTO_API_KEY} -e OPENAI_API_KEY=${OPENAI_API_KEY} -it robo-mo poetry run streamlit run robo_mo/chatbot.py
+
+build-common-ts:
+	cd infrastructure && npm ci --verbose && npm run build && cd ..
+
+pull-request-ci: build-common-ts lint docker-build
```
Comment (bruuuuuuuce) [inline Makefile:36]:
```suggestion
pull-request-ci: build-common-ts docker-build
```

File: Makefile [hunk 5]
```diff
@@ -16,8 +24,14 @@ run-chatbot:
...
 
 .PHONY: docker-run
 docker-run:
 	@docker run -p 8501:8501 -e MOMENTO_API_KEY=${MOMENTO_API_KEY} -e OPENAI_API_KEY=${OPENAI_API_KEY} -it robo-mo poetry run streamlit run robo_mo/chatbot.py
+
+build-common-ts:
+	cd infrastructure && npm ci --verbose && npm run build && cd ..
+
+pull-request-ci: build-common-ts docker-build
+	MOCK_LOOKUPS_FOR_CI_SYNTH=true npx cdk --app "npx ts-node --prefer-ts-exts bin/infrastructure.ts" synth
```
Comment (bruuuuuuuce) [inline Makefile:37]:
```suggestion
	cd infrastructure && MOCK_LOOKUPS_FOR_CI_SYNTH=true npx cdk --app "npx ts-node --prefer-ts-exts bin/infrastructure.ts" synth
```

File: Makefile [hunk 6]
```diff
@@ -16,8 +24,14 @@ run-chatbot:
...
 
 .PHONY: docker-run
 docker-run:
 	@docker run -p 8501:8501 -e MOMENTO_API_KEY=${MOMENTO_API_KEY} -e OPENAI_API_KEY=${OPENAI_API_KEY} -it robo-mo poetry run streamlit run robo_mo/chatbot.py
+
+build-common-ts:
+	cd infrastructure && npm ci --verbose && npm run build && cd ..
+
+pull-request-ci: build-common-ts docker-build
+	cd infrastructure && MOCK_LOOKUPS_FOR_CI_SYNTH=true npx cdk --app "npx ts-node --prefer-ts-exts bin/infrastructure.ts" synth
```
Comment (bruuuuuuuce) [inline Makefile:37]:
```suggestion
	cd infrastructure && MVI_CHAT_DOMAIN=sample.com MOCK_LOOKUPS_FOR_CI_SYNTH=true npx cdk --app "npx ts-node --prefer-ts-exts bin/infrastructure.ts" synth
```

File: Makefile [hunk 7]
```diff
@@ -16,8 +24,14 @@ run-chatbot:
...
 
 .PHONY: docker-run
 docker-run:
 	@docker run -p 8501:8501 -e MOMENTO_API_KEY=${MOMENTO_API_KEY} -e OPENAI_API_KEY=${OPENAI_API_KEY} -it robo-mo poetry run streamlit run robo_mo/chatbot.py
+
+build-common-ts:
+	cd infrastructure && npm ci --verbose && npm run build && cd ..
+
+pull-request-ci: build-common-ts docker-build
+	cd infrastructure && MVI_CHAT_DOMAIN=sample.com MOCK_LOOKUPS_FOR_CI_SYNTH=true npx cdk --app "npx ts-node --prefer-ts-exts bin/infrastructure.ts" synth
```
Comment (malandis) [inline Makefile:37]: Is this what we want for `MVI_CHAT_DOMAIN`?

**Review by cprice404 — APPROVED** [review]

**Pull request merged**